### PR TITLE
feat: Remove config-seed and add environment overrides

### PR DESCRIPTION
**Pull request by lenny-intel**
Addresses #189

This `draft` PR is dependent on https://github.com/edgexfoundry/edgex-go/pull/2445 being merged first and all the docker images created in Nexus.

### CODE REVIEW

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml [hunk 1]
```diff
@@ -20,14 +20,22 @@ version: '3.4'
...
   Logging_EnableRemote: "true"
   Databases_Primary_Type: mongodb
   Databases_Primary_Host: edgex-mongo
   Databases_Primary_Port: 27017
-  support-logging_Writable_Persistence: database
+  SecretStore_Host: edgex-vault
+  SecretStore_ServerName: edgex-vault
+  SecretStore_RootCaCertPath: /tmp/edgex/secrets/ca/ca.pem
+  # Value `none` required since compose doesn't all can't environment value of empty string
+  SecretStore_Authentication_AuthToken: none
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml]: What happens if this is removed entirely?  There is no real-life scenario where this value will ever be statically seeded.

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml [hunk 2]
```diff
@@ -255,6 +244,15 @@ services:
...
         aliases:
             - edgex-proxy
+    environment:
+      <<: *common-variables
+      KongURL_Server: kong
+      SecretService_Server: edgex-vault
+      SecretService_SecretService_TokenPath: /tmp/edgex/secrets/edgex-security-proxy-setup/secrets-token.json
+      CACertPath: /tmp/edgex/secrets/ca/ca.pem
+      SecretService_SNIS: "edgex-kong"
+      Clients_RulesEngine_Host: edgex-support-rulesengine
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml]:
```suggestion
```

edgex-proxy doesn't have a dependency on either rules engine or or device-virtual.  These can be removed entirely.

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml [hunk 3]
```diff
@@ -255,6 +244,15 @@ services:
...
+      Clients_VirtualDevice_Host: edgex-device-virtual
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml]:
```suggestion
```

edgex-proxy doesn't have a dependency on either rules engine or or device-virtual. These can be removed entirely.

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml [hunk 4]
```diff
@@ -20,14 +20,22 @@ version: '3.4'
...
   Logging_EnableRemote: "true"
   Databases_Primary_Type: mongodb
   Databases_Primary_Host: edgex-mongo
   Databases_Primary_Port: 27017
-  support-logging_Writable_Persistence: database
+  SecretStore_Host: edgex-vault
+  SecretStore_ServerName: edgex-vault
+  SecretStore_RootCaCertPath: /tmp/edgex/secrets/ca/ca.pem
+  # Value `none` required since compose doesn't all can't environment value of empty string
+  SecretStore_Authentication_AuthToken: none
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml]: Ditto

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml [hunk 5]
```diff
@@ -255,6 +244,15 @@ services:
...
             - edgex-proxy
+    environment:
+      <<: *common-variables
+      KongURL_Server: kong
+      SecretService_Server: edgex-vault
+      SecretService_SecretService_TokenPath: /tmp/edgex/secrets/edgex-security-proxy-setup/secrets-token.json
+      CACertPath: /tmp/edgex/secrets/ca/ca.pem
+      SecretService_SNIS: "edgex-kong"
+      Clients_RulesEngine_Host: edgex-support-rulesengine
+      Clients_VirtualDevice_Host: edgex-device-virtual
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml]: Ditto

**Comment by bnevis-i** [comment]
releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml
releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml

mongo: Needs overrride for SecretStore_TokenFile

**Comment by bnevis-i** [comment]
End of preliminary review -- did not spend a lot of time reviewing the non-security compose file changes.

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml (hunk 1 above)
Comment (lenny-intel) [reply releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml]: @bnevis-i In the non-docker configuration.toml it is set to `egdex`. This is required to override that value. If `AuthToken` can be removed from the non-docker, then the override here will not be require.

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml (hunk 3 above)
Comment (lenny-intel) [reply releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml]: Great!

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml (hunk 4 above)
Comment (lenny-intel) [reply releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml]: @bnevis-i , Ditto my response.. ;-)

**Comment by lenny-intel** [comment]
> releases/nightly-build/compose-files/docker-compose-nexus-mongo.yml
> releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml
> 
> mongo: Needs overrride for SecretStore_TokenFile

Our custom Mongo Docker image doesn't have environment overrides and isn't changed by the current PRs . Since it is a Docker images, it should just have the single configuration for Docker and not require overrides. Make sense?

**Comment by lenny-intel** [comment]
> End of preliminary review -- did not spend a lot of time reviewing the non-security compose file changes.

Thanks!

File: releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml (hunk 1 above)
Comment (bnevis-i) [reply releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml]: @tonyespy   Are you expecting to `AuthToken: edgex` in any of the snap transform scripts?

Comment (bnevis-i) [reply releases/nightly-build/compose-files/docker-compose-nexus-mongo-arm64.yml]: @tonyespy NM.  Checked. It isn't.

**Comment by bnevis-i** [comment]
> Our custom Mongo Docker image doesn't have environment overrides and isn't changed by the current PRs . Since it is a Docker images, it should just have the single configuration for Docker and not require overrides. Make sense?

Makes sense but we need to use caution as snaps often transform the standard configuration.toml for snap usage even though it is not using mongo in a container (however, the snap is probably just going to cut over to Redis I would expect).  It is ok to kick this can down the road.  Just warning that it is not safe to delete the docker/configuration.toml out of edgex-mongo at this time.

**Comment by lenny-intel** [comment]
> It is ok to kick this can down the road. Just warning that it is not safe to delete the docker/configuration.toml out of edgex-mongo at this time.

Agreed. Focus is on edgex-go. This can be revisited at a later time.

**Comment by bnevis-i** [comment]
@lenny-intel could you add a depends-on logging dependency to edgex-proxy?

**Comment by lenny-intel** [comment]
@bnevis-i , I added more overrides to Proxy to disable the RemoteLogging set in the global overrides.

**Pull request merged**